src/main/java/ServerSocketManager.java

```diff
@@ -1,24 +1,26 @@
 import java.io.*;
 import java.net.ServerSocket;
 import java.net.Socket;
-import java.util.Scanner;
 
 public class ServerSocketManager implements SocketManager {
 
     private Socket socket;
+    private CommandLineInterface cli;
+
+    public ServerSocketManager(CommandLineInterface cli) {
+        this.cli = cli;
+    }
 
     public void connect(int port) throws IOException {
         ServerSocket serverSocket = new ServerSocket(port);
         socket = serverSocket.accept();
     }
 
     public String receiveString() throws IOException {
-        Scanner serverInput = new Scanner(socket.getInputStream());
-        return serverInput.nextLine();
+         return cli.getInput(socket.getInputStream());
     }
 
     public void present(String message) throws IOException {
-        PrintWriter serverOutput = new PrintWriter(new OutputStreamWriter(socket.getOutputStream()), true);
-        serverOutput.println(message);
+        cli.getOutput(socket.getOutputStream(), message);
```

Comment:
Is this method getting output from the CLI or sending output to be displayed?

Reply:
This method gets the output from the output stream and then displays it to the client via the CLI - is this method doing too much?

Apologies for the delayed response, coming back to this now!